clientupdate/clientupdate.go

```diff
@@ -259,13 +269,17 @@ func Update(args Arguments) error {
 }
 
 func (up *Updater) confirm(ver string) bool {
-	switch cmpver.Compare(version.Short(), ver) {
-	case 0:
-		up.Logf("already running %v version %v; no update needed", up.Track, ver)
-		return false
-	case 1:
-		up.Logf("installed %v version %v is newer than the latest available version %v; no update needed", up.Track, version.Short(), ver)
-		return false
+	// Only check version when we're not switching tracks.
+	if up.Track == "" || up.Track == CurrentTrack {
+		c := cmpver.Compare(up.currentVersion, ver)
```

Comment:
Fine as written, but in case you didn't know about it, switch also allows a short decl:

```go
switch c := cmpver.Compare(up.currentVersion, ver); {
case c == 0:
// etc.
```

Reply:
I did not know that! Today is a good day.